src/offset_vector.c

```diff
@@ -40,26 +42,32 @@ uint32_t _ovi_Next(void *ctx, RSQueryTerm **t);
 void _ovi_Rewind(void *ctx);
 
 /* memory pool for buffer iterators */
-static mempool_t *__offsetIters = NULL;
-static mempool_t *__aggregateIters = NULL;
+static pthread_key_t __offsetIters;
+static pthread_key_t __aggregateIters;
+
+static void __attribute__((constructor)) initKeys() {
+  pthread_key_create(&__offsetIters, (void(*)(void*))mempool_destroy);
+  pthread_key_create(&__aggregateIters, (void(*)(void*))mempool_destroy);
+}
 
 /* Free it */
 void _ovi_free(void *ctx) {
-  mempool_release(__offsetIters, ctx);
+  mempool_release(pthread_getspecific(__offsetIters), ctx);
 }
 
 void *newOffsetIterator() {
   return rm_malloc(sizeof(_RSOffsetVectorIterator));
 }
 /* Create an offset iterator interface  from a raw offset vector */
 RSOffsetIterator RSOffsetVector_Iterate(const RSOffsetVector *v, RSQueryTerm *t) {
-  if (!__offsetIters) {
+  mempool_t *pool = pthread_getspecific(__offsetIters);
```

Comment:
You are getting/setting in the local thread dictionary. Any impacts on the main thread when multithreading is off? Do we need a special case here?